### ISSUE TITLE
Potential errors in the project

### ISSUE THREAD

**Issue by NikiPn** [bitwarden/server#2013]
### Steps To Reproduce

Hello!

I’d like to show you several issues found in your project.

### Potential error in [BillingInfo.cs](https://github.com/bitwarden/server/blob/master/src/Core/Models/Business/BillingInfo.cs)
The property is assigned values twice.

Below is a code fragment with this issue:
```
    public BillingInvoice(Invoice inv)
    {
         Amount = inv.AmountDue / 100M;    // <=
         Date = inv.Created;
         Url = inv.HostedInvoiceUrl;
         PdfUrl = inv.InvoicePdf;
         Number = inv.Number;
         Paid = inv.Paid;
         Amount = inv.Total / 100M;        // <=
    }
```

The result of the inv.AmountDue / 100 expression is [assigned](https://github.com/bitwarden/server/blob/master/src/Core/Models/Business/BillingInfo.cs#L142) to Amout. However, there’s a similar operation five lines below but this time inv.Total / 100M is [assigned](https://github.com/bitwarden/server/blob/master/src/Core/Models/Business/BillingInfo.cs#L148).

### Potential error in [AppleIapService.cs](https://github.com/bitwarden/server/blob/master/src/Core/Services/Implementations/AppleIapService.cs).
Perhaps the '??' operator doesn’t work as expected.

Here’s this code fragment:
```
    throw new Exception("Failed verifying Apple IAP after too many attempts. Last attempt status: " +
    lastReceiptStatus?.Status ?? "null");
```

The '??' operator’s priority is lower than the '+' operator’s priority. Therefore, the value of the Status property is added to the string first, and after that the null-coalescing operator snaps into action.

[Link](https://github.com/bitwarden/server/blob/master/src/Core/Services/Implementations/AppleIapService.cs#L95) to the line with error.

### Potential error in [FreshdeskController.cs](https://github.com/bitwarden/server/blob/master/src/Billing/Controllers/FreshdeskController.cs).
Erroneous postfix notation.

The fragment with a potential error:
```
    private async Task<HttpResponseMessage> CallFreshdeskApiAsync(...., int retriedCount = 0)
    {
         ....
         return await CallFreshdeskApiAsync(request, retriedCount++);             // <=
    }
```

During the postfix incrementation the current value of the variable is returned first and only after that this value is increased.

[Link](https://github.com/bitwarden/server/blob/master/src/Billing/Controllers/FreshdeskController.cs#L167) to the line with error.

All the described issues were found when we checked your project with the PVS-Studio analyzer. You can find more issues in the [article](https://pvs-studio.com/en/blog/posts/csharp/0947/). 

### Expected Result

No errors in the code.

### Actual Result

The presence of errors in the code.

### Build Version

1.48.1

### Environment

Cloud (bitwarden.com)

**Comment by justindbaur** [comment]
Hi @NikiPn this is a really great write-up and accompanying article. I'm going to assign this to me and slowly start tackling them but I also don't want to steal your thunder. You did a lot of awesome ground work here and I would be very open to reviewing some pull requests if you want to make them. 

At the beginning of the year we started enforcing linting rules in our CI pipeline and I have a goal of extending that by both being stricter with the built in analyzers and eventually enabling nullable annotations. Once we've got that foundation set I am all for building on top of it and looking at outside analyzers so that we can keep our code quality top notch. Once again, thanks a ton!

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
